.github/workflows/generate-speakeasy-on-pr.yaml

```diff
@@ -0,0 +1,78 @@
+name: Run Speakeasy on PR Open
+
+on:
+  pull_request:
+    paths-ignore:
+      - '.github/**'
+      - 'examples/**'
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Branch to run the workflow on'
+        required: true
+        default: 'feat/platform-api-automated-update'
+
+permissions:
+  contents: read
```

Comment:
Will the PR be updated by this action after running `make speakeasy`?
In that case, wouldn't the gh action need write permission?

Reply:
It would but for now we just want to make sure that it doesn't OOM. So I'll update the permissions then.